File: Mediator/PHP/LockingSystem.php
<?php

require_once "Command.php";

class LockingSystem {
    private bool $status;
    private Command $command;

    public function __construct()
    {
        $this->status = false;
    }

    public function lock(Command $command) {
        $this->command = $command;
        $this->status = true;
    }

    public function unlock() {
        $this->status = false;
    }

    public function checkStatus() {
        if ($this->status) {
            echo "The system currently used by:\n";
            $this->command->describe();
        } 

        return $this->status;
    }
}

File: Mediator/Python 3/LockingSystem.py
class LockingSystem:
  def __init__(self):
    self.status = False
    self.command = None
  
  def lock(self, command):
    self.status = True
    self.command = command
  
  def unlock(self):
    self.status = False
    self.command = None
  
  def checkStatus(self):
    if self.status:
      print("The system is currently used by:")
      self.command.describe()
    
    return self.status